model/__init__.py

```diff
@@ -328,6 +328,29 @@ def secrets_server(self, cfg_name=None):
     def protecode(self, cfg_name=None):
         return self.cfg_factory.protecode(self._default_name('protecode', cfg_name))
 
+    def _cfg_element_names(self, cfg_type_name: str):
+        ''' Return an iterator yielding the names of all config-elements for a given ConfigType
+        in this ConfigurationSet
+        '''
+        ensure_not_empty(cfg_type_name)
+
+        types = self._cfg_types()
+        if not cfg_type_name in types:
```

Comment:
remove this

instead: collect all config_names for the given cfg_type_name or the empty list / set / iterator

in addition: check if cfg_type is known by our factory; fail in this case